youtube_dl/extractor/adn.py

```diff
@@ -132,6 +179,14 @@ def _real_extract(self, url):
                 urljoin(self._BASE_URL, links_url), video_id, headers={
                     'Authorization': 'Bearer ' + authorization,
                 })
+            if links_data.get('status'):
+                if links_data.get('code') == 1:
+                    self.raise_login_required(links_data.get('error'))
+                if links_data.get('code') == 0:
+                    if links_data.get('error').find('Pour des raisons l') != -1:
```

Comment:
use `in` operator and move `error = links_data.get('error')` statement instead of using `links_data.get('error')` multiple times and check for error availibility before using it.

Reply:
in case of error ADN always returns: {status, code, error}

Reply:
what is true for now, can differ in the future, so it's better to make the extraction process more robust.

Reply:
It's hard for me to believe in this eventuality, but I changed test to `error`

I'm think that this is something like this: `raise HttpForbidden(msg, code)`